pkg/wasm/keychain/keychain.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keychain
+
+import (
+	"time"
+
+	"istio.io/istio/pkg/bootstrap/platform"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+var registeredKeychains = map[platform.PlatformType]authn.Keychain{
+	platform.PlatformTypeNone: authn.DefaultKeychain,
+}
+
+// Registers the given key chain with the platform type.
+// To take account for the default keychain always, prepend the default keychain.
+func RegisterKeychain(t platform.PlatformType, k authn.Keychain) {
+	registeredKeychains[t] = authn.NewMultiKeychain(authn.DefaultKeychain, k)
+}
+
+// Returns a key chain with the support for vendor specific keychain by the given platform type.
+func GetPlatformSpecificKeyChain(platformType platform.PlatformType) authn.Keychain {
```

Comment:
I was thinking could this be a method of env in pkg/bootsreap/platform. 
so we can get rid of implementing another platform for ketchain, maybe later some other function also for platforms

Reply:
Make sense. Done to change as you explained.